File: web/src/api/yunguan/product/productSupply.js
import request from "@/utils/request";

// 添加系统运行工具
export function addSupplyInfo(modelData) {
    return request({
        url: "product/supply/addSupplyInfo",
        method: "post",
        data: modelData,
    });
}

// 修改系统运行工具
export function updateSupplyInfo(data) {
    return request({
        url: "product/supply/updateSupplyInfo",
        method: "put",
        data: data,
    });
}

// 获取系统运行工具列表
export function getSupplyInfoList(params) {
    return request({
        url: "product/supply/querySupplyInfo",
        method: "post",
        data: params,
    });
}

//同步单个数据
export function SyncSupplyDemandInfo(params) {
    return request({
        url: 'product/supply/syncSupplyInfo',
        method: "post",
        data: params
    })
}

// 删除系统运行工具
export function deleteSupplyInfo(ID) {
    return request({
        url: "product/supply/deleteSupplyInfo",
        method: "delete",
        data: { ID: ID },
    });
}

export const createProduct = (data) => {
    return request({
        url: '/product/supply/createProduct',
        method: 'post',
        data
    })
}

export const deleteProduct = (params) => {
    return request({
        url: '/product/supply/deleteProduct',
        method: 'delete',
        params
    })
}

export const deleteProductByIds = (params) => {
    return request({
        url: '/product/supply/deleteProductByIds',
        method: 'delete',
        params
    })
}

export const getSupplyOrDemandCityList = () => {
    return request({
        url: '/infoRegion/getInfoRegionList',
        method: 'get',
    })
}
//获取配置列表
export const getProductConfigList = (params) => {
    return request({
        url: '/sysType/getSysTypeList',
        method: 'get',
        params: params
    })
}

export const getProductList = (params) => {
    return request({
        url: '/product/supply/getProductList2',
        method: 'get',
        params: params
    })
}
//供需编辑接口
export const updateProduct = (data) => {
    return request({
        url: '/product/supply/updateProduct',
        method: 'put',
        data
    })
}

File: server/router/yunguan/product/productSupply.go
package product

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type ProductSupplyRouter struct {
}

func (s *ProductSupplyRouter) InitProductSupplyRouter(Router *gin.RouterGroup) {
	productRouter := Router.Group("/")

	dataRouter := Router.Group("product/supply")

	supplyInfoApi := v1.ApiGroupApp.YunGuanApiGroup.ProductSupplyApi

	{
		productRouter.GET("sysType/getSysTypeList", supplyInfoApi.GetSupplyConfig)         //获取系统类型
		productRouter.GET("infoRegion/getInfoRegionList", supplyInfoApi.GetSupplyCityList) //获取区域信息
	}
	{
		dataRouter.POST("addSupplyInfo", supplyInfoApi.AddSupplyInfo)            // 添加数据
		dataRouter.PUT("updateSupplyInfo", supplyInfoApi.UpdateSupplyInfo)       // 修改数据
		dataRouter.DELETE("deleteSupplyInfo", supplyInfoApi.DeleteSupplyInfo)    // 删除数据
		dataRouter.POST("querySupplyInfo", supplyInfoApi.QuerySupplyInfo)        // 获取数据列表
		dataRouter.POST("syncSupplyInfo", supplyInfoApi.SyncSupplyDemandInfo)    // 同步单个数据
		dataRouter.POST("createProduct", supplyInfoApi.CreateSupplyInfo)         // 创建供需
		dataRouter.DELETE("deleteProduct", supplyInfoApi.DeleteSupplyInfo)       //删除供需
		dataRouter.DELETE("deleteProductByIds", supplyInfoApi.DeleteSupplyByIds) //删除供需
		dataRouter.GET("getProductList2", supplyInfoApi.GetSupplyList)           //获取供需
		dataRouter.PUT("updateProduct", supplyInfoApi.UpdateSupplyInfo)          //供需更新
	}
}
